services/system/Accounts/src/RAccounts.cpp

```diff
@@ -16,50 +17,23 @@ using Tables = psibase::ServiceTables<psibase::WebContentTable>;
 
 namespace SystemService
 {
-   std::optional<HttpReply> RAccounts::serveSys(HttpRequest request)
+   struct AccountsQuery
    {
-      auto to_json = [](const auto& obj)
-      {
-         auto json = psio::convert_to_json(obj);
-         return HttpReply{
-             .contentType = "application/json",
-             .body        = {json.begin(), json.end()},
-         };
-      };
-
-      if (request.method == "GET")
-      {
-         // TODO: replace with GraphQL
-         // if (request.target.starts_with("/api/accounts"))
-         // {
-         //    auto accountsTables = Accounts::Tables(Accounts::service);
-         //    auto accountTable     = accountsTables.open<AccountTable>();
-         //    auto accountIndex     = accountTable.getIndex<0>();
-
-         //    std::vector<Account> rows;
-         //    for (auto it = accountIndex.begin(); it != accountIndex.end(); ++it)
-         //       rows.push_back(*it);
-         //    return to_json(rows);
-         // }
-
-         // TODO: replace with GraphQL
-         if (request.target == "/accounts")
-         {
-            auto accountsTables = Accounts::Tables(Accounts::service);
-            auto accountTable   = accountsTables.open<AccountTable>();
-            auto accountIndex   = accountTable.getIndex<0>();
-
-            std::vector<Account> rows;
-            for (auto it = accountIndex.begin(); it != accountIndex.end(); ++it)
-               rows.push_back(*it);
-            return to_json(rows);
-         }
+      auto exists(AccountNumber account) const
```

Comment:
Instead of a boolean, I'd prefer a query that returns the Account or null if it doesn't exist.

Reply:
Done